src/System.Linq/src/System/Linq/Enumerable.Select.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        public static IEnumerable<TResult> Select<TSource, TResult>(this IEnumerable<TSource> source, Func<TSource, TResult> selector)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            if (selector == null) throw Error.ArgumentNull("selector");
+            if (source is Iterator<TSource>) return ((Iterator<TSource>)source).Select(selector);
+            if (source is TSource[]) return new SelectArrayIterator<TSource, TResult>((TSource[])source, selector);
+            if (source is List<TSource>) return new SelectListIterator<TSource, TResult>((List<TSource>)source, selector);
+            if (source is ICollection<TSource>) return new SelectCollectionIterator<TSource, TResult>((ICollection<TSource>)source, selector);
```

Comment:
Per the [style guidelines](https://github.com/dotnet/corefx/wiki/Coding-style), braces may be omitted but the block must be properly indented on its own line.